kv/memberlist/memberlist_client.go

```diff
@@ -347,8 +355,14 @@ func NewKV(cfg KVConfig, logger log.Logger, dnsProvider DNSProvider, registerer
 		codecs:         make(map[string]codec.Codec),
 		watchers:       make(map[string][]chan string),
 		prefixWatchers: make(map[string][]chan string),
-		shutdown:       make(chan struct{}),
-		maxCasRetries:  maxCasRetries,
+		notifyCh:       make(chan *bytes.Buffer, notifyMsgQueueSize),
+		notifyBuffPool: sync.Pool{
+			New: func() interface{} {
+				return &bytes.Buffer{}
+			},
+		},
+		shutdown:      make(chan struct{}),
+		maxCasRetries: maxCasRetries,
```

Comment:
if `maxCasRetries` is a global constant, why does it also need to be a property of `KV` instead of just directly accessing it?

Reply:
We change it in the tests, but don't want to expose it as option to clients.

Reply:
maybe it would be worth it to add a comment explaining that then

Reply:
there's already a [comment](https://github.com/grafana/dskit/blob/main/kv/memberlist/memberlist_client.go#L280-L283) in place specifying that the purpose of `maxCasRetries` field is to alter KV behavior when running tests